### ISSUE TITLE
Cannot add commands with "reserved" names: getAlias, reset

### ISSUE THREAD

**Issue by alastair-todd** [cypress-io/cypress#6146]
```js
Cypress.Commands.add('getAlias', (alias:string) => cy.get(`@${alias}`));
```

results in 

> TypeError: command.get is not a function

but

```js
Cypress.Commands.add('alias', (alias:string) => cy.get(`@${alias}`));
```

does not

what _trickerypokery_?

**Comment by jennifer-shehane** [comment]
I can recreate this behavior.

```js
Cypress.Commands.add('getAlias', (alias:string) => cy.get(`@${alias}`));

cy.getAlias('foo')
```

The command above throws the error `TypeError: command.get is not a function` when called

<img width="999" alt="Screen Shot 2020-01-13 at 11 41 01 AM" src="https://user-images.githubusercontent.com/1271364/72234080-99b31280-35f9-11ea-934a-12096ed6d080.png">


This error is throwing from this line of code as shown from the stack trace you can see when you click on the yellow error: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/querying.js#L240:L240

I thought this may have been some sort of regression with a decaffeinated file but this error throws back to version 3.4.1.

**Comment by jennifer-shehane** [comment]
Another instance of a 'reserved' word where adding the command with the name 'reset' does very weird things from https://github.com/cypress-io/cypress/issues/7482

https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress.js#L364:L364

```js
Cypress.Commands.add('reset', () => {})

describe('tests', () => {
  before(() => {})

  it('test', () => {
    expect(true).to.be.true
  })
})
```

**Comment by ATMartin** [comment]
:wave: I've got a draft up for a possible solution here @ https://github.com/cypress-io/cypress/pull/8732 - will make sure to update the docs accordingly as well!

**Comment by maddhruv** [comment]
Hey @jennifer-shehane what should be the ideal solution for issues like this?
- For sure allowing the users to customize or override the reserved/internal commands can result in random issues
- Moreover, if there was a way, whenever cypress registers the custom commands, it can throw an error/warning that the command name is reserved or something (eslint-plugin can be a great start)

**Comment by chrisbreiding** [comment]
The code for this is done (addressed in #18923), but has yet to be released. We'll update this issue and reference the changelog when it's released.
